internal/controller/pkg/revision/runtime.go

```diff
@@ -68,6 +68,9 @@ var (
 
 // ManifestBuilder builds the runtime manifests for a package revision.
 type ManifestBuilder interface {
+	// ReferencedServiceAccountName returns the name of the referenced service
+	// account that already exist, typically managed externally.
+	ReferencedServiceAccountName() string
```

Comment:
Nit: I find interfaces like this a little easier to read when there is whitespace between the method and the following comment string. Like the one immediately below, for example.

Reply:
The method seems a little out of place in this interface, where all the other methods build and return manifests.

What does it method return if there isn't a referenced service account?

Reply:
Answering my own question - I see it returns the empty string if there isn't a referenced service account.

Reply:
> The method seems a little out of place in this interface, where all the other methods build and return manifests.

I'm wondering whether it's possible to avoid adding this method. Could you:

1. Update the `Deployment` implementation to not call `DeploymentWithServiceAccount` if `template.serviceAccountName` is already set. (Or update `DeploymentWithServiceAccount` to be a no-op in that case.)
2. Call `ServiceAccount` but not apply the resulting service account.
3. Call `Deployment` and check whether the `template.serviceAccountName` matches the service account's

The idea being that if the deployment has a `template.serviceAccountName` that isn't the service's we assume we don't need to create a service.

I could be pretty easily convinced that this is worse/more magic than what you're doing here - just thinking out loud to see whether there's a way to avoid this out-of-place seeming interface method.

Reply:
Thanks, I agree that your suggested flow is better. 
Updated accordingly.